File: robot/location.py
'''Uses sensor data to locate robot in pit and navigate the robot to locations
Processes image, lidar, and encoder data to find location
Also can automaticly preform basic operations such as scooping and dumping
Command usage of lidar unit though sensor.py
Obstical detection?
'''
import sensor
import remoteEvent

#Location of robot in the pit
pitLoc = remoteEvent.RemoteVarEvent("pitLoc", {"x": 0, "y": 0})

#Location the robot is going to
pitLocCom = remoteEvent.RemoteVarEvent("pitLocComand", {"x": 0, "y": 0})

#Preforms digging operation
locDoADig = remoteEvent.RemoteEvent("locDoADig")

#Cancels current location module operation
locCancel = remoteEvent.RemoteEvent("locCancel")

File: browserControl/js/location.js
/**
 * Commands to use with location module
 * @module control/location
 */
import {RemoteEvent, RemoteVarEvent} from "./remoteEvent.js";

/**Location of robot in the pit
 * @prop {float} x - X location
 * @prop {float} y - Y location
 */
const pitLoc = new RemoteVarEvent("pitLoc", {"x": 0, "y": 0})

/**Location the robot is going to
 * @prop {float} x - X location
 * @prop {float} y - Y location
 */
const pitLocCom = new RemoteVarEvent("pitLocComand", {"x": 0, "y": 0})

/**Preforms digging operation
 */
const locDoADig = new RemoteEvent("locDoADig")

/**When close to bin will move to proper postion and preform dumping operation
 */
const locDoADump = new RemoteEvent("locDoADump")

/**Cancels current location module operation
 */
const locCancel = new RemoteEvent("locCancel")

export {pitLoc, pitLocCom, locDoADig, locDoADump, locCancel}